burnman/data/input_raw_endmember_datasets/HPdata_to_burnman.py

```diff
@@ -74,74 +68,105 @@ def getmbr(ds, mbr):
-print 'File autogenerated using HPdata_to_burnman.py'
-print '"""'
-print ''
-print 'from ..mineral import Mineral'
-print 'from ..solidsolution import SolidSolution'
-print 'from ..solutionmodel import *'
-print 'from ..processchemistry import dictionarize_formula, formula_mass'
-print ''
-
-solutionfile = 'tc-ds62_solutions.txt'
-with open(solutionfile, 'r') as fin:
-    print fin.read()
-fin.close()
-
-print '"""'
-print 'ENDMEMBERS'
-print '"""'
-print ''
-formula = '0'
-for i in range(int(ds[0][0])):
-    mbr = ds[i * 4 + 3][0]
-    M = getmbr(ds, mbr)
-    if mbr == 'and':  # change silly abbreviation
-        mbr = 'andalusite'
-    if M.flag != -1 and M.flag != -2 and M.k[0] > 0:
-        print 'class', mbr, '(Mineral):'
-        print '    def __init__(self):'
-        print ''.join(['        formula=\'', M.formula, '\''])
-        print '        formula = dictionarize_formula(formula)'
-        print '        self.params = {'
-        print ''.join(['            \'name\': \'', M.name, '\','])
-        print '            \'formula\': formula,'
-        print '            \'equation_of_state\': \'hp_tmt\','
-        print '            \'H_0\':', M.H * 1e3, ','
-        print '            \'S_0\':', M.S * 1e3, ','
-        print '            \'V_0\':', M.V * 1e-5, ','
-        print '            \'Cp\':', [round(M.Cp[0] * 1e3, 10), round(M.Cp[1] * 1e3, 10), round(M.Cp[2] * 1e3, 10), round(M.Cp[3] * 1e3, 10)], ','
-        print '            \'a_0\':', M.a, ','
-        print '            \'K_0\':', M.k[0] * 1e8, ','
-        print '            \'Kprime_0\':', M.k[1], ','
-        print '            \'Kdprime_0\':', M.k[2] * 1e-8, ','
-        print '            \'n\': sum(formula.values()),'
-        print '            \'molar_mass\': formula_mass(formula)}'
-        if M.flag != 0:
-            print '        self.property_modifiers = [['
-        if M.flag == 1:
-            print '            \'landau_hp\', {\'P_0\':', 1.e5, ','
-            print '                          \'T_0\':', 298.15, ','
-            print '                          \'Tc_0\':', M.od[0], ','
-            print '                          \'S_D\':', M.od[1] * 1e3, ','
-            print '                          \'V_D\':', M.od[2] * 1e-5, '}]]'
-        if M.flag == 2:
-            print '            \'bragg_williams\', {\'deltaH\':', M.od[0] * 1e3, ','
-            print '                               \'deltaV\':', M.od[1] * 1e-5, ','
-            print '                               \'Wh\':', M.od[2] * 1e3, ','
-            print '                               \'Wv\':', M.od[3] * 1e-5, ','
-            print '                               \'n\':', M.od[4], ','
-            print '                               \'factor\':', M.od[5], '}]]'
-        print '        Mineral.__init__(self)'
-        print ''
-print ''
+with open('HP_2011_ds62.py', 'wb') as outfile:
+    outfile.write('# This file is part of BurnMan - a thermoelastic and '
```

Comment:
wouldn't it be cleaner if you use a python multiline string ``"""    """``?

Reply:
That makes for really ugly looking code (as I would not be able to indent consecutive lines - the multiline string includes any blank space before or after the code)

Reply:
(c.f. https://stackoverflow.com/questions/10660435/pythonic-way-to-create-a-long-multi-line-string)